src/main/java/org/reactome/qa/doisuggester/DOISuggesterMain.java

```diff
@@ -0,0 +1,172 @@
+						for (String rleKey : s.getRlesToInstanceEdits().keySet())
+						{
+							String instanceEdits = s.getRlesToInstanceEdits().get(rleKey).stream().reduce("", (prev, instEd) -> prev + " | " + instEd );
+							List<String> record = new ArrayList<>();
+							record.add(pathway);
+							record.add(rleKey);
+							record.add(instanceEdits);
+							records.add(record);
+							rleCountsForPathways.put(pathway, rleCountsForPathways.get(pathway) + 1);
+						}
+					}
+				}
+			}
+
+			int total = 0;
+			for (Entry<String, Integer> entry : rleCountsForPathways.entrySet() )
+			{
+				System.out.println(entry.getValue() + " RLEs for " + entry.getKey());
+				total += entry.getValue();
+			}
+			System.out.println( total + " total RLEs have DOI suggestion.");
+			printOutput(records);
+		}
+		catch (PropertyHasNoValueException | PropertyNotPresentException e)
+		{
+			System.err.println("Your properties file appears to be misconfigured: " + e.getMessage());
+			e.printStackTrace();
+			System.exit(-1);
+		}
+		catch (FileNotFoundException e)
+		{
+			System.err.println("The config.properties file could not be found. Now exiting.");
+			System.exit(-1);
+		}
+		catch (SQLException e)
+		{
+			System.err.println("SQL Exception caught - most likely there was a problem connecting to the database. Maybe check your connection settings and try again?");
+			e.printStackTrace();
+			System.exit(-1);
+		}
+		catch (InvalidAttributeException e)
+		{
+			System.err.println("Error retrieving data - an attempt was made to access data via an invalid attribute.");
+			e.printStackTrace();
+			System.exit(-1);
+		}
+		catch (Exception e)
+		{
+			e.printStackTrace();
+			System.exit(-1);
+		}
+		System.out.println("All done.");
+	}
+
+	/**
+	 * Print the suggestions to a file.
+	 * @param suggestions A list of lists - each sublist is an element of a suggestion.
+	 * @throws IOException - Could be thrown if there is a problem writing the output to a file.
+	 */
+	private static void printOutput(List<List<String>> records) throws IOException
+	{
+		final int pathwayIndex = 0;
+		final int rleIndex = 1;
+		final int instanceEditIndex = 2;
+		try (CSVPrinter printer = new CSVPrinter(new FileWriter(new File("doi-suggestions.csv")), CSVFormat.DEFAULT.withQuoteMode(QuoteMode.ALL).withHeader("Pathway","RLE", "InstanceEdits")))
+		{
+			// This comparator is used to ensure the outputs are neatly sorted by Pathway, and then by RLE.
+			//
+			// Since we're sorting the string representation of these objects, the output will actually end up being sorted by DB ID,
+			// since string representations are usually structured as something like "[${DB_ID}] ${_displayName}". But that's not a big deal - we just
+			// want *SOME* ordering imposed so records of different Pathways are all grouped together in the output. Otherwise, the report is harder
+			// to read.
+			Comparator<List<String>> recordComparator = (arg0, arg1) -> {
```

Comment:
Maybe for content it could be `record1` and `record2` instead of `arg0` and `arg1`?